packages/okta-adapter/src/dependency_changers/order_policies_by_priority.ts

```diff
@@ -0,0 +1,160 @@
+ * You may not use this file except in compliance with the License.  You may obtain a copy of the License at https://www.salto.io/terms-of-use
+ *
+ * CERTAIN THIRD PARTY SOFTWARE MAY BE CONTAINED IN PORTIONS OF THE SOFTWARE. See NOTICE FILE AT https://github.com/salto-io/salto/blob/main/NOTICES
+ */
+
+import _ from 'lodash'
+import {
+  Change,
+  DependencyChange,
+  DependencyChanger,
+  InstanceElement,
+  ReferenceExpression,
+  dependencyChange,
+  getChangeData,
+  isAdditionChange,
+  isInstanceChange,
+  isReferenceExpression,
+} from '@salto-io/adapter-api'
+import { deployment } from '@salto-io/adapter-components'
+import { getParent } from '@salto-io/adapter-utils'
+import { logger } from '@salto-io/logging'
+import { POLICY_PRIORITY_TYPE_NAMES, POLICY_RULE_PRIORITY_TYPE_NAMES } from '../constants'
+import { ALL_SUPPORTED_POLICY_NAMES, POLICY_RULE_TYPES_WITH_PRIORITY_INSTANCE } from '../filters/policy_priority'
+
+const log = logger(module)
+type ChangeWithKey = deployment.dependency.ChangeWithKey<Change<InstanceElement>>
+
+const createDependencyChange = (prev: ChangeWithKey, current: ChangeWithKey): DependencyChange[] => [
+  dependencyChange('add', current.key, prev.key),
+]
+
+const groupChangesByType = (changes: ChangeWithKey[]): Record<string, ChangeWithKey[]> =>
+  changes.reduce(
+    (acc, change) => {
+      const instance = getChangeData(change.change)
+      const { typeName } = instance.elemID
+      if (!acc[typeName]) {
+        acc[typeName] = []
+      }
+      acc[typeName].push(change)
+      return acc
+    },
+    {} as Record<string, ChangeWithKey[]>,
+  )
+
+const groupRuleChangesByPolicy = (changes: ChangeWithKey[]): Record<string, ChangeWithKey[]> =>
+  changes.reduce(
+    (acc, change) => {
+      const instance = getChangeData(change.change)
+      try {
+        const parent = getParent(instance)
+        const parentFullName = parent.elemID.getFullName()
+        if (!acc[parentFullName]) {
+          acc[parentFullName] = []
+        }
+        acc[parentFullName].push(change)
+      } catch (e) {
+        log.error('Failed to get parent for %s', instance.elemID.getFullName())
+      }
+      return acc
+    },
+    {} as Record<string, ChangeWithKey[]>,
+  )
+
+const getGroupedPoliciesAndRules = (
+  additionPolicyChanges: ChangeWithKey[],
+  additionRuleChanges: ChangeWithKey[],
+): {
+  policies: ChangeWithKey[][]
+  rules: ChangeWithKey[][]
+} => ({
+  policies: Object.values(groupChangesByType(additionPolicyChanges)),
+  rules: Object.values(groupRuleChangesByPolicy(additionRuleChanges)),
+})
+
+const sortByPriority = (group: ChangeWithKey[], priorityMap: Record<string, number>): ChangeWithKey[] =>
+  group.sort((change1, change2) => {
+    const priority1 = priorityMap[getChangeData(change1.change).elemID.getFullName()] ?? Infinity
+    const priority2 = priorityMap[getChangeData(change2.change).elemID.getFullName()] ?? Infinity
+    return priority1 - priority2
+  })
```

Comment:
I think this can be replaced with - 

```suggestion
const sortByPriority = (group: ChangeWithKey[], priorityMap: Record<string, number>): ChangeWithKey[] =>
  _.sortBy(
    group,
    change => priorityMap[getChangeData(change1.change).elemID.getFullName()] ?? Infinity
  )
```